kubernetes/kubernetes.go

```diff
@@ -186,6 +187,41 @@ func CollectKubeMetrics(config KubeAgentConfig) {
 
 }
 
+func performConnectionChecks(ka *KubeAgentConfig) {
+	log.Info("Performing connectivity checks. Checking that the agent can retrieve S3 URL")
+
+	cldyMetricClient, err := client.NewHTTPMetricClient(client.Configuration{
+		Token:         ka.APIKey,
+		Verbose:       false,
+		ProxyURL:      ka.OutboundProxyURL,
+		ProxyAuth:     ka.OutboundProxyAuth,
+		ProxyInsecure: ka.OutboundProxyInsecure,
+	})
+	if err != nil {
+		log.Fatalf("error creating Cloudability Metric client: %v ", err)
+	}
+
+	metricSampleURL := apiEndpoint + "/metricsample"
+
+	file, err := os.Create("/tmp/temp.txt")
+	if err != nil {
+		log.Warnf("Failed to create file temp.txt")
+	}
+
+	_, err = file.WriteString("Health Check")
+	if err != nil {
+		log.Warnf("Failed to write in file temp.txt")
+	}
+
+	_, _, err = cldyMetricClient.GetUploadURL(file, metricSampleURL, cldyVersion.VERSION, ka.clusterUID)
```

Comment:
Given that a customer could potentially have the upload domain blocked it could be kind of a convenient check to have it upload the tmp file as well assuming that it overwrites seamlessly.

I don't know if we've seen that specific issue occur previously, but I know we have seen general connectivity on the actual upload process several times.

Reply:
Yeah we have seen the issue where a customer can get the url but not upload however it is very rare. We also discussed this and don't want to send unnecessary file uploads to our s3. Could be something we revisit depending on the number of issues where uploading sees error when s3 url is fine.